classes/Mail.php

```diff
@@ -869,15 +869,26 @@ public static function mimeEncode($string, $charset = 'UTF-8', $newline = "\r\n"
      * @param string $to Email address
      *
      * @return string
+     *
+     * Try to use INTL_IDNA_VARIANT_UTS46 only if defined or PHP version >= 8.0.0
+     * If not, use INTL_IDNA_VARIANT_2003
+     * PHP 7.2.0 up to 8.0.0 will return E_DEPRECATED notices but INTL_IDNA_VARIANT_2003 will still work.
+     * From PHP 8.0.0, we cannot longer use INTL_IDNA_VARIANT_2003 so we'll get an Exception using default if INTL_IDNA_VARIANT_UTS46 not defined.
+     * See https://wiki.php.net/rfc/deprecate-and-remove-intl_idna_variant_2003
      */
     public static function toPunycode($to)
     {
         $address = explode('@', $to);
         if (empty($address[0]) || empty($address[1])) {
             return $to;
         }
-
-        return $address[0] . '@' . idn_to_ascii($address[1], 0, INTL_IDNA_VARIANT_UTS46);
+        if (defined('INTL_IDNA_VARIANT_UTS46')) {
```

Comment:
Because of return you can do:

```php
        if (defined('INTL_IDNA_VARIANT_UTS46')) {
            return $address[0] . '@' . idn_to_ascii($address[1], 0, INTL_IDNA_VARIANT_UTS46);
        } 

        if (version_compare(phpversion(), '8.0.0', '<')) {
            return $address[0] . '@' . idn_to_ascii($address[1], 0, INTL_IDNA_VARIANT_2003);
        } 

        return $address[0] . '@' . idn_to_ascii($address[1]);
```

Reply:
Hi @PierreRambaud
Thanks for the review.
You are right, ‘return’ will stop execution at the first true comparission so I guess elseif is not required.

Reply:
I did it for you + cs fixer =)
We really appreciate contribution with explanations and php docs :heart: